server/src/main/java/org/opensearch/search/builder/SearchSourceBuilder.java

```diff
@@ -297,6 +299,9 @@ public SearchSourceBuilder(StreamInput in) throws IOException {
                 derivedFields = in.readList(DerivedField::new);
             }
         }
+        if (in.getVersion().onOrAfter(Version.V_2_18_0)) {
+            searchPipeline = in.readOptionalString();
+        }
```

Comment:
You'll need to do the whole BWC dance, since 2.x is building 2.18.

1. Do this on main with `onOrAfter(Version.V_3_0_0))`.
2. You'll need a manual backport to 2.x, where you do `onOrAfter(Version.V_2_18_0)`.
3. Before merging the backport to 2.x, open another PR on main to change it to `onOrAfter(Version.V_2_18_0)`.
4. Merge the backport PR.
5. Merge the main version update PR.

We need to do those last two steps as close together as possible, as merging the backport PR will break builds on main.

I hate this process, but we haven't been able to think of something better.

Reply:
Yes, you are right. Been a while changed an API body. Pushed the change and also added one more test for serialization.